api/app/Extensions/Socialite/Two/ProviderTrait.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Extensions\Socialite\Two;
+
+use Cache;
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\Config;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+trait ProviderTrait
+{
+    /**
+     * Redirect the user of the application to the provider's authentication screen.
+     *
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function redirect()
+    {
+        $state = Str::random(40);
+
+        if ($this->usesState()) {
+            $this->request->getSession()->set('state', $state);
+        }
```

Comment:
When could this scenario occur? We're always running stateless/sessionless right?